### PR TITLE
Add pool maintainer fee in staking pool

### LINKED ISSUE #36: Support taking mining pool fee
...such that the a certain portion of miner rewards should go to the pool

### PR DESCRIPTION
**Pull request by ninjaahhh**
resolves #36

### CODE REVIEW

File: contracts/StakingPool.sol [hunk 1]
```diff
@@ -19,16 +19,34 @@ contract StakingPool {
...
+    uint256 public minerFeeRateBp;
     uint256 public minerReward;
-    uint256 public maxStakers;
 
-    constructor(address _miner, address _admin, uint256 _feeRateBp, uint256 _maxStakers) public {
-        require(_feeRateBp <= MAX_BP, "Fee rate should be in basis point.");
+    // Mining pool maintainer and corresponding fee structure.
+    address public poolMaintainer;
+    // Pool fee rate in basis point, note it's part of miner reward.
+    uint256 public poolMaintainerFeeRateBp;
```
Comment (qizhou) [inline contracts/StakingPool.sol]: In general, I feel a bit confused about minerFee and poolMaintainerFee from their naming.  Could we just put them  separately, and just required (poolMaintainerFeeRateBp + minerFeeRateBp <= MAX_BP)?

Comment (ninjaahhh) [reply contracts/StakingPool.sol]: makes sense. updated

**Comment by ninjaahhh** [comment]
updated as we discussed offline - now pool fee is the same as the miner fee. take a look when you have time @qizhou

File: contracts/StakingPool.sol [hunk 2]
```diff
@@ -72,32 +93,44 @@ contract StakingPool {
...
+        minerFeeRateBp = _minerFeeRateBp;
     }
 
     function calculateStakesWithDividend(address staker) public view returns (uint256) {
         if (totalStakes == 0) {
             return 0;
         }
         uint256 dividend = getDividend(address(this).balance);
-        uint256 stakerPayout = dividend.mul(MAX_BP - feeRateBp).div(MAX_BP);
+        uint256 stakerPayout = dividend.mul(MAX_BP - minerFeeRateBp).div(MAX_BP);
```
Comment (qizhou) [inline contracts/StakingPool.sol]: Need to decrease also maintainer's?

Comment (ninjaahhh) [reply contracts/StakingPool.sol]: done

**Pull request merged**